src/dmd/root/file.d

```diff
@@ -22,6 +22,320 @@ import dmd.root.filename;
 import dmd.root.rmem;
 import dmd.root.string;
 
+/**
+Runs a non-pure function or delegate as pure code. Use with caution.
+Returns: whatever `fun` returns.
+*/
+private auto fakePure(F)(scope F fun) pure
+{
+    return (cast(mixin(F.stringof ~ " pure")) fun)();
```

Comment:
This is the new `mixin("int") var;` feature, is it not? Surely older compilers don't grok this.

Reply:
Ha, didn't cross my mind. Will replace with a more conservative mixin.